test/event/handlers/CommandParserHandlerTest.ts

```diff
@@ -0,0 +1,128 @@
+import { expect } from "chai";
+import { Constants } from "discord.js";
+import { Message } from "discord.js";
+import { SinonSandbox, createSandbox } from "sinon";
+
+import { COMMAND_PREFIX, BOTLESS_CHANNELS } from "../../../src/config.json";
+import MockCommand from "../../MockCommand";
+import MockDiscord from "../../MockDiscord";
+import Command from "../../../src/abstracts/Command";
+import CommandFactory from "../../../src/factories/CommandFactory";
+import CommandParserHandler from "../../../src/event/handlers/CommandParserHandler";
+
+describe("CommandParserHandler", () => {
+	describe("constructor()", () => {
+		let sandbox: SinonSandbox;
+
+		beforeEach(() => {
+			sandbox = createSandbox();
+		});
+
+		it("creates a handler for MESSAGE_CREATE", () => {
+			const handler = new CommandParserHandler();
+
+			expect(handler.getEvent()).to.equal(Constants.Events.MESSAGE_CREATE);
+		});
+
+		it("should call loadCommands()", () => {
+			const factoryMock = sandbox.stub(CommandFactory.prototype, "loadCommands");
+
+			const handler = new CommandParserHandler();
+
+			expect(factoryMock.calledOnce).to.be.true;
+		});
+
+		afterEach(() => {
+			sandbox.restore();
+		});
+	});
+
+	describe("handle", () => {
+		let sandbox: SinonSandbox;
+		let handler: CommandParserHandler;
+		let discordMock: MockDiscord;
+		let command: MockCommand;
+
+		beforeEach(() => {
+			sandbox = createSandbox();
+			handler = new CommandParserHandler();
+			discordMock = new MockDiscord();
+			command = new MockCommand();
+		});
+
+		it("should not run command if it doesn't start with command prefix", async () => {
+			const runCommandMock = sandbox.stub(command, "run");
+
+			const message = discordMock.getMessage();
+
+			message.content = command.getName();
+
+			await handler.handle(message);
+
+			expect(runCommandMock.called).to.be.false;
+		});
+
+		it("should not run command if message was sent on a botless channel", async () => {
+			const runCommandMock = sandbox.stub(command, "run");
+
+			const message = discordMock.getMessage();
+			const botlessChannelId : string = Object.values(BOTLESS_CHANNELS)[0] as string;
```

Comment:
Can you remove the space before the `:` please?

Reply:
What happens if `BOTLESS_CHANNELS` is empty? Does this test fail?
Probably better just to make up an ID as we're not really doing integration tests anyway.

Reply:
Oops, I didn't think about that case.

The test passes, but for the wrong reason: the command isn't ran because it's not recognized by `CommandFactory`, so `commandExists()` returns `false`. In the other test cases I mock `commandExists()` so it returns `true`. I'm wondering if I should also mock `commandExists()` to return true in the prefix test to avoid this behavior.

When you say "make up an id", you mean adding a fake channel id to the `BOTLESS_CHANNELS` object in `config.json`? That seems like a good workaround to me.

Reply:
Nah, don't add the fake id to the `BOTLESS_CHANNELS` object, just hard code it in the test.

Reply:
If I hardcode the channel ID, this expression will evaluate to `false`, making the test fail:
`Object.values(BOTLESS_CHANNELS).includes("my-hardcoded-channel-id")`

Reply:
Oh, I see - my bad. Is there a way to mock the variable? If not I guess we'll just have to do it with a value in the `config.json`.

Reply:
Sadly, Sinon only let's us stub method calls, so we wouldn't be able to stub the export value of `config.json`. But apparently there's a NPM package called `proxyquire` that let's you do this (override dependencies). 

Do you think it would be interesting to use it, or should we stay with the mock channel thing? The latter seems a bit inelegant in my opinion.

https://stackoverflow.com/questions/56890285/how-to-stub-a-json-file-with-sinon

Reply:
Do whatever you thinks best, I don't mind! :)